doc/content/design/numa.md

```diff
@@ -0,0 +1,129 @@
+# NUMA
+
+NUMA stands for Non-Uniform Memory Access and describes that RAM access
+for CPUs in a multi-CPU system is not equally fast for all of them. CPUs
+are grouped into so-called nodes and each node has fast access to RAM
+that is considered local to its node and slower access to other RAM. In
+the context of  CPU virtualisation assigning vCPUs to NUMA nodes is an
+optimisation strategy to reduce memory latency. This document describes
+a design to make NUMA-related assignments for domains (hence, VMs)
+visible to the user. Below we refer to these assignments and
+optimisations collectively as NUMA for simplicity.
+
+## NUMA Properties
+
+Xen in XenServer 9 (XS 9) implements NUMA optimisation. We want to
+expose the following NUMA-related properties of VMs to API clients, and
+in particualar XenCenter. Each one is represented by a new field in
+XAPI's VM data model:
+
+* RO `VM.numa_optimised`: boolean: if the VM is optimised for NUMA 
+* RO `VM.numa_nodes`: integer: number of NUMA nodes of the host the VM
+  is using
+* MRO `VM.numa_node_memory`: list of tuple (`node_X`: integer,
+  `amount_in_bytes`: int64): amount of VM memory in NUMA node `X`.
+
+NUMA support is only available in XenServer 9. For now, I would suggest
+to make the required changes in XenServer 8 as well but no VM there
+will be NUMA optimised. Some parts of the code will have to be managed
+by patches.
```

Comment:
Xapi has been able to produce NUMA-optimised VMs for at least 2 years for hosts configured to do so. Without patches in some cases it will fail to create  NUMA-optimised VMs, but I don't see a technical reason to prevent xapi on `master` from showing these stats